<launch>
  <!-- Launch the robot state publisher -->
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" output="screen">
    <param name="robot_description" command="$(find xacro)/xacro '$(find new_description)/urdf/new.xacro'" />
  </node>

  <!-- Launch RViz -->
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find new_description)/rviz_config.rviz" />

  <!-- Map server -->
  <node name="map_server" pkg="map_server" type="map_server" args="$(find new_description)/maps/bug_map.yaml" />

  <!-- AMCL -->
  <node name="amcl" pkg="amcl" type="amcl" output="screen">
    <param name="use_map_topic" value="true"/>
    <param name="base_frame_id" value="base_link"/>
    <param name="odom_frame_id" value="odom"/>
    <param name="scan_topic" value="/scan"/>
  </node>

  <!-- Navigation -->
  <node name="move_base" pkg="move_base" type="move_base" output="screen">
    <rosparam file="$(find new_description)/config/move_base_params.yaml" command="load"/>
  </node>

  <!-- Algorithm Node -->
  <node name="bug0v5_algorithm" pkg="new_description" type="bug0v5.py" output="screen" launch-prefix="python3"/>
</launch>
